### PR TITLE
feat(perception_utils): add new package

### PR DESCRIPTION
**Pull request by takayuki5168**
Signed-off-by: Takayuki Murooka <takayuki5168@gmail.com>

## Description

added perception_utils almost copied from the perception packages.

~This PR depends on, and contains the following PR. So ignore these changes (boost_polygon_utils) in this PR.~
https://github.com/autowarefoundation/autoware.universe/pull/1367

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by takayuki5168** [comment]
@yukkysaito @miursh 
Could you review this PR?
I've added the test.

This PR depends on, and contains the following PR. So ignore these changes (boost_polygon_utils) in this PR.
https://github.com/autowarefoundation/autoware.universe/pull/1367

**Comment by takayuki5168** [comment]
(I'll change Tier IV typo (- > TIER IV) with another PR since a lot of files use Tier IV)

File: common/perception_utils/include/perception_utils/perception_utils.hpp [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+#include <vector>
+
+namespace perception_utils
+{
+std::uint8_t getHighestProbLabel(
+  const std::vector<autoware_auto_perception_msgs::msg::ObjectClassification> & classification);
+
+autoware_auto_perception_msgs::msg::TrackedObject toTrackedObject(
+  const autoware_auto_perception_msgs::msg::DetectedObject & detected_object);
+
```
Comment (yukkysaito) [inline common/perception_utils/include/perception_utils/perception_utils.hpp:34]: If you are ok, can you add the following functions?
- [toDetectedObject](https://github.com/autowarefoundation/autoware.universe/blob/main/perception/detection_by_tracker/src/detection_by_tracker_core.cpp#L102-L121)
- transformObjects
  - https://github.com/autowarefoundation/autoware.universe/blob/main/perception/multi_object_tracker/src/multi_object_tracker_core.cpp#L58
  - https://github.com/autowarefoundation/autoware.universe/blob/main/perception/detection_by_tracker/src/detection_by_tracker_core.cpp#L48

Comment (takayuki5168) [reply common/perception_utils/include/perception_utils/perception_utils.hpp:71]: https://github.com/autowarefoundation/autoware.universe/pull/1377/commits/97b68835e75f0c49eb95a4b162357583b0b1d1b0
DONE

Comment (yukkysaito) [reply common/perception_utils/include/perception_utils/perception_utils.hpp:71]: Perfect :+1:

File: common/perception_utils/include/perception_utils/geometry.hpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
+// Copyright 2022 Tier IV, Inc.
```
Comment (yukkysaito) [inline common/perception_utils/include/perception_utils/geometry.hpp:1]:
```suggestion
// Copyright 2022 TIER IV, Inc.
```
same comment for other files :pray:

Comment (takayuki5168) [reply common/perception_utils/include/perception_utils/geometry.hpp:1]: image
I was planning to deal with it with another PR, but thanks!

Comment (takayuki5168) [reply common/perception_utils/include/perception_utils/geometry.hpp]: Give me lgtm!

**Review by yukkysaito — APPROVED** [review]
LGTM

**Pull request merged**